web/src/lib/utils/auth.ts

```diff
@@ -17,21 +19,34 @@ export const getAuthUser = async () => {
   }
 };
 
+const isAuthenticated = (): boolean => {
+  if (browser) {
+    const cookies = document.cookie.split('; ');
+    for (const cookie of cookies) {
+      const [name] = cookie.split('=');
+      if (name === 'immich_is_authenticated') {
+        return true;
+      }
+    }
+  }
+  return false;
+};
+
 export const authenticate = async (options?: AuthOptions) => {
   options = options || {};
-
   const savedUser = getSavedUser();
-  const user = savedUser || (await getAuthUser());
-
-  if (!user) {
-    redirect(302, AppRoute.AUTH_LOGIN);
-  }
+  const user = savedUser || (isAuthenticated() ? await getAuthUser() : null);
+  if (!options.public) {
```

Comment:
I think you might be able to just add one new condition at the top that is "if public && !user then return" and leave everything else the same.

Reply:
Or, inside of !user either do the redirect or not, depending if the route is public. Either way you should not do anything else doing down if the user is null.